src/transformers/models/pop2piano/modeling_pop2piano.py

```diff
@@ -0,0 +1,1707 @@
+                module.wi_0.bias.data.zero_()
+            module.wi_1.weight.data.normal_(mean=0.0, std=factor * ((self.config.d_model) ** -0.5))
+            if hasattr(module.wi_1, "bias") and module.wi_1.bias is not None:
+                module.wi_1.bias.data.zero_()
+            module.wo.weight.data.normal_(mean=0.0, std=factor * ((self.config.d_ff) ** -0.5))
+            if hasattr(module.wo, "bias") and module.wo.bias is not None:
+                module.wo.bias.data.zero_()
+        elif isinstance(module, Pop2PianoAttention):
+            # Mesh TensorFlow attention initialization to avoid scaling before softmax
+            # See https://github.com/tensorflow/mesh/blob/fa19d69eafc9a482aff0b59ddd96b025c0cb207d/mesh_tensorflow/transformer/attention.py#L136
+            d_model = self.config.d_model
+            key_value_proj_dim = self.config.d_kv
+            n_heads = self.config.num_heads
+            module.q.weight.data.normal_(mean=0.0, std=factor * ((d_model * key_value_proj_dim) ** -0.5))
+            module.k.weight.data.normal_(mean=0.0, std=factor * (d_model**-0.5))
+            module.v.weight.data.normal_(mean=0.0, std=factor * (d_model**-0.5))
+            module.o.weight.data.normal_(mean=0.0, std=factor * ((n_heads * key_value_proj_dim) ** -0.5))
+            if module.has_relative_attention_bias:
+                module.relative_attention_bias.weight.data.normal_(mean=0.0, std=factor * ((d_model) ** -0.5))
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, (Pop2PianoAttention, Pop2PianoStack)):
+            module.gradient_checkpointing = value
+
+    def _shift_right(self, input_ids):
+        decoder_start_token_id = self.config.decoder_start_token_id
+        pad_token_id = self.config.pad_token_id
+
+        if decoder_start_token_id is None:
+            raise ValueError(
+                "self.model.config.decoder_start_token_id has to be defined. In Pop2Piano it is usually set to the pad_token_id."
+            )
+
+        # shift inputs to the right
+        if is_torch_fx_proxy(input_ids):
+            # Item assignment is not supported natively for proxies.
+            shifted_input_ids = torch.full(input_ids.shape[:-1] + (1,), decoder_start_token_id)
+            shifted_input_ids = torch.cat([shifted_input_ids, input_ids[..., :-1]], dim=-1)
+        else:
+            shifted_input_ids = input_ids.new_zeros(input_ids.shape)
+            shifted_input_ids[..., 1:] = input_ids[..., :-1].clone()
+            shifted_input_ids[..., 0] = decoder_start_token_id
+
+        if pad_token_id is None:
+            raise ValueError("self.model.config.pad_token_id has to be defined.")
+        # replace possible -100 values in labels by `pad_token_id`
+        shifted_input_ids.masked_fill_(shifted_input_ids == -100, pad_token_id)
+
+        return shifted_input_ids
+
+
+@dataclass
+class Pop2PianoGreedySearchEncoderDecoderOutput(ModelOutput):
```

Comment:
Sorry - not sure why we need to override the usual model output data classes? As far as I can see, they are identical to the default ones? E.g. https://github.com/huggingface/transformers/blob/6b7d6f848bffb2ed68935858e71395c4070e3200/src/transformers/generation/utils.py#L179

Reply:
Actually the main difference is that`Pop2PianoGreedySearchEncoderDecoderOutput` enables us to output List 
``sequences: List[torch.LongTensor] = None``. 

In `GreedySearchEencoderDecoderOutput` (as the docs says) the sequences must have the shape (batch_size, sequence_length), but since for Pop2Piano each example gives us tensors of shape (dim1,  dim2) where for each example dim1 and dim2 values are different! I could have padded those values again but  I preferred to output them as list and thus I had to rewrite the class(s) as the usual model output data classes only allow  torch.LongTenesor.

Reply:
Ok thanks for the explanation! Still not convinced we need the list output, since any generations that terminate early will be padded with the EOS token. Take the example of text generation. If we generate the sequences:
```
This is the first sentence in the batch.
This is the second sentence in the batch, which is two words longer.
```

Then our predictions will be padded to max length by the `.generate` method, where any early terminations get padded with the EOS token:
```
This is the first sentence in the batch. <|endoftext|> <|endoftext|> <|endoftext|> <|endoftext|> <|endoftext|>
This is the second sentence in the batch, which is five words longer. <|endoftext|>
```

Now they have the same length, so we can just output a single tensor of shape (`batch_size`, `seq_length`)

To tackle the issue of padding, we usually pass the argument `skip_special_tokens=True` to the tokenizer to omit any EOS tokens from the generated sequences (see the example https://huggingface.co/openai/whisper-large-v2#english-to-english)

I think we should be able to achieve the same design here?